openedx/core/djangoapps/team_api/views.py

```diff
@@ -0,0 +1,407 @@
+
+    def get_queryset(self):
+        """
+        Returns the queryset used to access the given team.
+        """
+        return CourseTeam.objects.all()
+
+
+class TopicListView(APIView):
+    """HTTP endpoint for retreiving a list of topics for a given course."""
+
+    authentication_classes = (SessionAuthenticationAllowInactiveUser,)
+    permission_classes = (permissions.IsAuthenticated,)
+
+    page_size = 10
+    # TODO determine the correct max page size
+    max_page_size = sys.maxint
+
+    def get(self, request):
+        """
+        GET /api/team/v0/topics/?course_id={course_id}
+        """
+        try:
+            course_id_string = request.QUERY_PARAMS.get('course_id', None)
+            if course_id_string is None:
+                return Response(status=status.HTTP_400_BAD_REQUEST)
+
+            course_id = CourseKey.from_string(course_id_string)
+
+            course_module = modulestore().get_course(course_id)
+            if course_module is None:  # course is None if not found
+                return Response(status=status.HTTP_404_NOT_FOUND)
+
+            if CourseEnrollment.get_enrollment(request.user, course_id) is None:
+                return Response({'detail': "user must be enrolled"}, status=status.HTTP_403_FORBIDDEN)
+
+            topics = [t for t in course_module.teams_topics if t['is_active']]
+
+            if 'text_search' in request.QUERY_PARAMS:
+                return Response({'detail': "text_search is not yet supported"}, status=status.HTTP_400_BAD_REQUEST)
+
+            ordering = request.QUERY_PARAMS.get('order_by', 'name')
+            if ordering == 'name':
+                topics = sorted(topics, key=lambda t: t['name'])
+            elif ordering == 'team_count':
+                topics = sorted(topics, cmp=lambda t1, t2: t1['team_count'] > t2['team_count'])
+            else:
+                return Response({'detail': "unsupported order_by value {}".format(ordering)},
+                                status=status.HTTP_400_BAD_REQUEST)
+
+            if 'page_size' in request.QUERY_PARAMS:
+                self.page_size = min(self.max_page_size, int(request.QUERY_PARAMS['page_size']))
+
+            paginator = Paginator(topics, self.page_size)
+            page = paginator.page(request.QUERY_PARAMS.get('page', 1))
+            serializer = PaginationSerializer(instance=page)
+            return Response(serializer.data)  # May be None
+        except InvalidKeyError:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+
+class TopicDetailView(APIView):
+    """HTTP endpoint for retreiving details of a particular topic."""
+
+    authentication_classes = (SessionAuthenticationAllowInactiveUser,)
+    permission_classes = (permissions.IsAuthenticated,)
+
+    def get(self, request, topic_id, course_id):
+        """
+        GET /api/team/v0/topics/{topic_id},{course_id}/
+        """
+        try:
+            course_id = CourseKey.from_string(course_id)
```

Comment:
Looks like there is some common code here that could be shared with lines 338+ above.